### PR TITLE
kernel-uki: remove usrmount from initrd

### PR DESCRIPTION
**Pull request by jozzsi**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [ ] The toolchain has been rebuilt successfully (or no changes were made to it)
- [ ] The toolchain/worker package manifests are up-to-date
- [ ] Any updated packages successfully build (or no packages were changed)
- [ ] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [ ] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [ ] All package sources are available
- [ ] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [ ] LICENSE-MAP files are up-to-date (`./LICENSES-AND-NOTICES/SPECS/data/licenses.json`, `./LICENSES-AND-NOTICES/SPECS/LICENSES-MAP.md`, `./LICENSES-AND-NOTICES/SPECS/LICENSE-EXCEPTIONS.PHOTON`)
- [ ] All source files have up-to-date hashes in the `*.signatures.json` files
- [ ] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [ ] Documentation has been updated to match any changes to the build system
- [ ] Ready to merge

---

###### Summary 

When systemd is included in initrd usrmount is not needed as it is a no-operation. This PR should slightly increase initrd generation and help reasoning about the list of dracut modules.

See https://github.com/dracut-ng/dracut-ng/blob/main/modules.d/98usrmount/module-setup.sh#L16

###### Does this affect the toolchain?  

**YES**

CC @trungams @ddstreetmicrosoft

### CODE REVIEW

**Review by trungams — CHANGES_REQUESTED** [review]
Hi @jozzsi, thank you for your PR. I've verified the change works. 

For package changes, we require updating the checksum file (if applicable) and version bumping the relevant packages. To save time from tedious work, you can cherry-pick commit f91c38d7cfcf24226f73ef5d90bd6acfb555db64 where I have already made those changes, then provide your name + email in the changelog lines of the updated spec files, e.g [here](https://github.com/microsoft/azurelinux/commit/f91c38d7cfcf24226f73ef5d90bd6acfb555db64#diff-612fa9d7bfec1778446855c261c82b48f4860dba556f3a515d004d408b5f967dR148). You don't have to use your real name.

Once you make the remaining change, I'll be happy to take this PR.

**Review by trungams — COMMENTED** [review]
Just one comment. And one more thing I overlooked: could you switch the target branch to `3.0-dev` instead of `3.0`? The latter is our stable branch only used for release. Thank you

**Comment by jozzsi** [comment]
> Just one comment. And one more thing I overlooked: could you switch the target branch to `3.0-dev` instead of `3.0`? 

Screenshot 2024-08-29 at 19 06 06

Is the target not 3.0-dev ?

**Comment by trungams** [comment]
It was 3.0 when I wrote the comment, then I changed it to 3.0-dev (didn't know I can do that for other's PR), but seems like there is some extra change now due to the switch

**Comment by jozzsi** [comment]
Thanks @trungams . I updated the PR just to be one commit.

File: SPECS/kernel/kernel.spec [hunk 1]
```diff
@@ -407,6 +407,9 @@ echo "initrd of kernel %{uname_r} removed" >&2
 %{_sysconfdir}/bash_completion.d/bpftool
 
 %changelog
+* Thu Aug 29 2024 <> - 6.6.47.1-2
```
Comment (trungams) [inline SPECS/kernel/kernel.spec:410]: Missing name and email here

**Review by trungams — COMMENTED** [review]
Just one last change request here 🙏

File: SPECS/kernel/kernel.spec (hunk 1 above)
Comment (jozzsi) [reply SPECS/kernel/kernel.spec]: Fixed. Sorry.

**Review by trungams — APPROVED** [review]
LGTM. Thank you for the contribution :)

**Review by ms-mahuber — APPROVED** [review]

**Review by christopherco — APPROVED** [review]

**Comment by trungams** [comment]
@jozzsi Looks like there's a merge conflict. Could you help me resolve those? Also, can you reply to the bot above regarding the CLA?

**Comment by jozzsi** [comment]
@microsoft-github-policy-service agree

**Review by PawelWMS — APPROVED** [review]

**Pull request merged**